nncf/sparsity/magnitude/algo.py

```diff
@@ -49,36 +49,41 @@ def __init__(self, target_model: NNCFNetwork,
 
     def statistics(self):
         stats = super().statistics()
-        stats['sparsity_threshold'] = self._select_threshold(self.sparsity_rate_for_sparsified_modules)
+        stats['sparsity_threshold'] =\
+             self._select_threshold(self.sparsity_rate_for_sparsified_modules, self.sparsified_module_info)
         return stats
 
     def freeze(self):
         pass
 
     def set_sparsity_level(self, sparsity_level):
+        self.set_sparsity_level_for_module(sparsity_level, self.sparsified_module_info)
```

Comment:
Now this isn't right. Before this change, our sparsity algorithms set a *global* sparsity level; 60% would mean that 60% of total parameters in sparsified modules would be set to zero, but each module may have a local, per-module sparsity rate other than 60%. Now you've changed it so that setting a 60% sparsity level will always set 60% sparsity locally for each module, both in the global sparsity scenario and in the per-layer sparsity scenario, which is wrong. Fix this so that the current configs still preserve original behaviour - apply *global* sparsity only, not necessarily uniform across the modules (layers).

Reply:
Also, once you are done, provide a statistics table from the console log after running one of NNCF's magnitude sparsity training before the change and after the change, so that we can be sure that the configs we distribute (which don't use per-layer sparsity) still produce the same sparsity distribution as before. It is strange that our existing unit tests did not catch this.

Reply:
Sorry, I guess I misread the code here. Remove the `self.sparsified_module_info` from the signature - you probably don't need to pass an attribute to the method of the same class, just access it in the method. Everywhere where you pass a sparsified modules info list, rename the arguments appropriately: `self.sparsified_module_info` -> `self.sparsified_modules_info_list`. Same with the function - `set_sparsity_level_for_module` seems to be accepting a list of module**s**, and not setting a sparsity level for a single module - either refactor the function, or rename it, or do both.

Reply:
I can't access `self.sparsified_module_info` in the method because the method is used in local sparsity (where we pass a list of one module).  The second argument to  `self.set_sparsity_level_for_module`  is not always `self.sparsified_module_info`. 
In order not to pass its attribute to a class method, I can make  the method as `static`.

Reply:
@asenina this looks wrong and it probably *is* wrong, and can be done much better. Rename the functions properly so that the names reflect what the functions do, rethink the function signatures and level of access (public/protected) and I'm sure that the correct way will present itself on its own.

Reply:
Done.